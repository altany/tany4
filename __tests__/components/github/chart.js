import { render } from '@testing-library/react'

it('renders a chart with data and no error unchanged', () => {
    const data=[{language:'Javascript', value:1}, {language:'CSS', value:2}]
    const error=undefined
    jest.mock('swr', ()=>({
        __esModule: true,
        default: ()=>({data, error})
    }))

    const Chart = require('../../../components/github/chart').default
    const { container } = render(<Chart />)
    expect(container).toMatchSnapshot()
})

it('renders a repo with an error unchanged', () => {
    const data=undefined
    const error='test error'
    jest.mock('swr', ()=>({
        __esModule: true,
        default: ()=>({data, error})
    }))

    const Chart = require('../../../components/github/chart').default
    const { container } = render(<Chart />)
    expect(container).toMatchSnapshot()
})

it('renders a repo with no data and no error unchanged', () => {
    const data=undefined
    const error=undefined
    jest.mock('swr', ()=>({
        __esModule: true,
        default: ()=>({data, error})
    }))

    const Chart = require('../../../components/github/chart').default
    const { container } = render(<Chart />)
    expect(container).toMatchSnapshot()
})