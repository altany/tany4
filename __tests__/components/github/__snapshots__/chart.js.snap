// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a chart with data and no error unchanged 1`] = `
<div>
  <div
    class="chartContainer"
  >
    <div
      class="chart"
      title="Click on any section to see a list of repos that uses the language"
    >
      <mock-react-resize-detector>
        <div
          class="recharts-responsive-container"
          style="width: 94%; height: 100%;"
        />
      </mock-react-resize-detector>
    </div>
    <div
      class="repoInfo"
      id="repos"
    />
  </div>
</div>
`;

exports[`renders a repo with an error unchanged 1`] = `
<div>
  <div>
    failed to load chart
  </div>
</div>
`;

exports[`renders a repo with no data and no error unchanged 1`] = `
<div>
  <div>
    loading chart data...
  </div>
</div>
`;
